### DISCUSSION TITLE
Scrolling of Map placed in ScollView does not work.

### DISCUSSION THREAD

**Discussion by cat0363** [dotnet/maui#13613] [Q&A]
To reproduce, layout as follows.

    <ScrollView Orientation="Vertical">
        <Grid RowDefinitions="100,100,100,100,*,100">
            <Grid Grid.Row="0" BackgroundColor="Red" />
            <Grid Grid.Row="1" BackgroundColor="Yellow" />
            <Grid Grid.Row="2" BackgroundColor="Green" />
            <Grid Grid.Row="3" BackgroundColor="Blue" />
            <Map Grid.Row="4" x:Name="map" HeightRequest="250" WidthRequest="250" VerticalOptions="Start" HorizontalOptions="Start" />
            <Grid Grid.Row="5" BackgroundColor="Purple" />
        </Grid>
    </ScrollView>

After scrolling to the Map, when I try to scroll on the map, horizontal scrolling 
works, but vertical scrolling does not. It seems that the tap event is being handled 
by the Map's parent control, ScrollView. I want to scroll the map when scrolling on 
the map placed in the ScrollView. 

Since the build environment has not been prepared, it cannot be confirmed 
for iOS, but at least it is as described for Android. 

https://user-images.githubusercontent.com/125236133/222052238-ebfa8077-e7fb-47bd-bc0f-95008c7d8904.mp4

The only solution I found is below.

    public override bool DispatchTouchEvent(MotionEvent e) 
    {
        if (e.Action == MotionEventActions.Down ||
            e.Action == MotionEventActions.Move ||
            e.Action == MotionEventActions.Up) 
        {
            // Search MapView Control
            var maps = (Window.DecorView as ViewGroup).GetChildrenOfType<MapView>();

            foreach (var map in maps) 
            {
                // Initialize MapView screen position 
                int[] pos = new int[2];
                // Set MapView screen position 
                map.GetLocationOnScreen(pos);
                // Create hit test rectangle 
                Rect hitRect = new Rect(pos[0], pos[1], map.Width, map.Height);
                // Judge hit test
                bool isHitTest = hitRect.Contains(e.GetX(), e.GetY());
                // Hit test result is OK
                if (isHitTest) 
                {
                    // Touch event intercept
                    map.Parent.RequestDisallowInterceptTouchEvent(true);
                }
            }
        }

        return base.DispatchTouchEvent(e);
    }

In the MainActivity.cs file, re-implement DispatchTouchEvent, get the MapView that exists 
in the page, execute the hit test, and if it is a MapView, intercept the touch event. 

https://user-images.githubusercontent.com/125236133/222053763-555131bf-e39b-4adb-bfd6-baf4c9969204.mp4

This works as expected, any other good ideas?
I have not confirmed how it works on iOS, so I would like to know if there is no problem with iOS.
Thank you.

**Comment by AlbrechtWM** [comment]
Was having the same darn problem.  Thank you for the workaround.  I had been trying custom maphandlers, deprecated renderers and all manner of things to try to get it to behave properly.

**Reply by cat0363** [reply]
Hi, @AlbrechtWM 
I'm glad it was helpful for you.I raised this issue as 
Issue [#13628](https://github.com/dotnet/maui/issues/13628). I would like to continue the discussion 
in Issue [#13628](https://github.com/dotnet/maui/issues/13628).
